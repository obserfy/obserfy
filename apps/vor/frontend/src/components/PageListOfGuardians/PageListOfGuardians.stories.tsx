import { FC } from "react"
import PageListOfGuardians from "./PageListOfGuardians"

export default {
  title: "Core/PageListOfGuardians",
  component: PageListOfGuardians,
  parameters: {
    componentSubtitle: "Just a simple PageListOfGuardians",
  },
}

export const Basic: FC = () => <PageListOfGuardians />
