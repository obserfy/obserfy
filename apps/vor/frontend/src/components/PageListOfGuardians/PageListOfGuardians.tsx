import { t, Trans } from "@lingui/macro"
import { FC, useState } from "react"
import { Box, Button, Card, Flex } from "theme-ui"
import { borderTop } from "../../border"
import { useGetSchoolGuardians } from "../../hooks/api/guardians/useGetSchoolGuardians"
import { ReactComponent as ChevronRight } from "../../icons/next-arrow.svg"
import {
  ADMIN_URL,
  GUARDIAN_PROFILE_URL,
  NEW_GUARDIAN_ADMIN_URL,
} from "../../routes"
import Icon from "../Icon/Icon"
import { Link } from "../Link/Link"
import SearchBar from "../SearchBar/SearchBar"
import TopBar, { breadCrumb } from "../TopBar/TopBar"
import Typography from "../Typography/Typography"

export const PageListOfGuardians: FC = () => {
  const guardians = useGetSchoolGuardians()
  const [searchTerm, setSearchTerm] = useState("")
  const filteredGuardians = guardians.data?.filter((guardian) =>
    guardian.name.match(new RegExp(searchTerm, "i"))
  )

  return (
    <Box sx={{ maxWidth: "maxWidth.lg" }} mx="auto" pb={5}>
      <TopBar
        breadcrumbs={[
          breadCrumb(t`Admin`, ADMIN_URL),
          breadCrumb(t`All Guardians`),
        ]}
      />

      <Card mx={[0, 3]} sx={{ borderRadius: [0, "default"] }}>
        <Flex sx={{ alignItems: "center" }}>
          <Typography.H6 m={3}>
            <Trans>All Guardians</Trans>
          </Typography.H6>

          <Link to={NEW_GUARDIAN_ADMIN_URL} sx={{ ml: "auto", mr: 3 }}>
            <Button variant="text" data-cy="addGuardian">
              <Trans>Create New</Trans>
            </Button>
          </Link>
        </Flex>

        <Flex p={3} pt={0}>
          <SearchBar
            value={searchTerm}
            onChange={(e) => setSearchTerm(e.target.value)}
            sx={{ backgroundColor: "background" }}
          />
        </Flex>
        {filteredGuardians?.map(({ id, name, email, phone }) => (
          <GuardianCard
            key={id}
            guardianId={id}
            name={name}
            email={email}
            phone={phone}
          />
        ))}
      </Card>
    </Box>
  )
}

const GuardianCard: FC<{
  guardianId: string
  name: string
  email?: string
  phone?: string
}> = ({ guardianId, name, email, phone }) => {
  return (
    <Link to={GUARDIAN_PROFILE_URL(guardianId)} sx={{ display: "block" }}>
      <Flex
        p={3}
        sx={{
          ...borderTop,
          alignItems: "center",
          transition: "background-color 100ms ease-in-out",
          "&:hover": {
            backgroundColor: "primaryLightest",
          },
        }}
      >
        <Typography.Body sx={{ width: "100%" }}>{name}</Typography.Body>
        <Typography.Body
          py={1}
          mx={2}
          px={email ? 0 : 2}
          backgroundColor={email ? "transparent" : "tintWarning"}
          sx={{
            width: "100%",
            borderRadius: "default",
            fontWeight: email ? "normal" : "bold",
            overflow: "hidden",
            textOverflow: "ellipsis",
            color: "textMediumEmphasis",
          }}
        >
          {email || <Trans>No email set</Trans>}
        </Typography.Body>

        <Typography.Body
          mx={2}
          py={1}
          px={phone ? 0 : 2}
          backgroundColor={phone ? "transparent" : "tintWarning"}
          sx={{
            display: ["none", "block"],
            width: "100%",
            borderRadius: "default",
            fontWeight: phone ? "normal" : "bold",
            overflow: "hidden",
            textOverflow: "ellipsis",
            color: "textMediumEmphasis",
          }}
        >
          {phone || <Trans>No phone set</Trans>}
        </Typography.Body>

        <Icon as={ChevronRight} ml={2} />
      </Flex>
    </Link>
  )
}

export default PageListOfGuardians
