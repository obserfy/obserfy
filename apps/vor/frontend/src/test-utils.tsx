/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
