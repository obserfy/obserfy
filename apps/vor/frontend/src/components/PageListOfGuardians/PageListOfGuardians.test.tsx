import { render } from "../../test-utils"
import { Basic } from "./PageListOfGuardians.stories"

describe("PageListOfGuardians", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
