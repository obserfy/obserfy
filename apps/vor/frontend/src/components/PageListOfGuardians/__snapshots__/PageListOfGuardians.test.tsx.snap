// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageListOfGuardians should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  padding-bottom: 64px;
  max-width: 900px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  margin: 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  border: none;
  padding: 4px;
}

.emotion-4:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-4:hover,
.emotion-4:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
  fill: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow-x: auto;
}

.emotion-7 {
  display: inline-block;
  font-size: 13px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  white-space: nowrap;
}

.emotion-7:last-child {
  color: var(--theme-ui-colors-textPrimary);
}

.emotion-7:last-child>.splitter {
  display: none;
}

.emotion-8:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-9 {
  margin-left: 4px;
  margin-right: 4px;
}

.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: inline-block;
  font-size: 13px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  white-space: nowrap;
}

.emotion-10:last-child {
  color: var(--theme-ui-colors-textPrimary);
}

.emotion-10:last-child>.splitter {
  display: none;
}

.emotion-11 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-surface);
  border-radius: 6px;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 0px 1px,rgba(0, 0, 0, 0.27) 0px 2px 4px -2px;
  margin-left: 0;
  margin-right: 0;
  border-radius: 0;
}

@media screen and (min-width: 40em) {
  .emotion-11 {
    margin-left: 16px;
    margin-right: 16px;
  }
}

@media screen and (min-width: 40em) {
  .emotion-11 {
    border-radius: 6px;
  }
}

.emotion-12 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-13 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: bold;
  line-height: heading;
  color: var(--theme-ui-colors-text);
  text-rendering: optimizeLegibility;
  font-size: 20px;
  margin: 16px;
}

.emotion-14 {
  margin-left: auto;
  margin-right: 16px;
}

.emotion-15 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  border: none;
}

.emotion-15:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-15:hover,
.emotion-15:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-16 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 16px;
  padding-top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-17 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: var(--theme-ui-colors-textMediumEmphasis);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 13px;
}

.emotion-18 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-19 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  margin-right: -33px;
  margin-left: 8px;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
  width: 24px;
  height: 24px;
  z-index: 2;
}

.emotion-20 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  width: auto;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 12px;
  color: var(--theme-ui-colors-text);
  padding-top: 8px;
  padding-bottom: 8px;
  background-color: var(--theme-ui-colors-surface);
  --theme-ui-input-autofill-bg: var(--theme-ui-colors-background);
  padding-left: 40px;
  font-size: 14px;
  width: 100%;
  width: 100%;
  background-color: var(--theme-ui-colors-background);
}

.emotion-20:autofill,
.emotion-20:autofill:hover,
.emotion-20:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-20:autofill:first-line,
.emotion-20:autofill:hover:first-line,
.emotion-20:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-20:-webkit-autofill,
.emotion-20:-webkit-autofill:hover,
.emotion-20:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-20:-webkit-autofill:first-line,
.emotion-20:-webkit-autofill:hover:first-line,
.emotion-20:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-20:hover,
.emotion-20:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-20::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-20::-moz-placeholder {
  font-style: italic;
}

.emotion-20:-ms-input-placeholder {
  font-style: italic;
}

.emotion-20::placeholder {
  font-style: italic;
}

.emotion-20:disabled {
  opacity: 0.68;
  border-color: var(--theme-ui-colors-border);
  color: var(--theme-ui-colors-textDisabled);
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <a
          class="emotion-3"
          data-cy="back-button"
          href="/dashboard/admin"
        >
          <button
            class="emotion-4"
          >
            <div
              class="emotion-5"
            />
          </button>
        </a>
        <div
          class="emotion-6"
        >
          <a
            class="emotion-7"
            href="/dashboard/admin"
          >
            <span
              class="emotion-8"
            >
              Admin
            </span>
            <span
              class="splitter emotion-9"
            >
              /
            </span>
          </a>
          <div
            class="emotion-10"
          >
            All Guardians
            <span
              class="splitter"
            >
              /
            </span>
          </div>
        </div>
      </div>
      <div
        class="emotion-11"
      >
        <div
          class="emotion-12"
        >
          <h6
            class="emotion-13"
          >
            All Guardians
          </h6>
          <a
            class="emotion-14"
            href="/dashboard/admin/guardians/new"
          >
            <button
              class="emotion-15"
              data-cy="addGuardian"
            >
              Create New
            </button>
          </a>
        </div>
        <div
          class="emotion-16"
        >
          <label
            class="emotion-17"
          >
            <div
              class="emotion-18"
            >
              <div
                class="emotion-19"
              />
              <input
                class="emotion-20"
                name="name"
                placeholder="Search"
                value=""
              />
            </div>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
